### PR TITLE
Make caclient not require `mut`

### PR DESCRIPTION
**Pull request by howardjohn**
No need for exclusive access here

### CODE REVIEW

**Review by adiprerepa — APPROVED** [review]
lgtm

File: src/identity/manager.rs [hunk 1]
```diff
@@ -184,12 +181,9 @@ impl<T: CertificateProvider + Clone + Send + 'static> CertificateProvider for Se
                             let (tx, rx) = watch::channel(None);
                             write_locked_cache.insert(id.clone(), rx.clone());
                             drop(write_locked_cache);
-                            tokio::spawn(SecretManager::refresh_handler(
-                                id.clone(),
-                                self.cache.clone(),
-                                self.client.clone(),
-                                tx,
-                            ));
+                            let s = self.clone();
```
Comment (hzxuzhonghu) [inline src/identity/manager.rs:184]: Doesn't this clone bring much cost

Comment (howardjohn) [reply src/identity/manager.rs:184]: No, it is designed to be cheap to clone (I also added a comment in the code to clarify this). It is just incrementing a reference counter

Comment (howardjohn) [reply src/identity/manager.rs:184]: Also what tonic recommends

Comment (howardjohn) [reply src/identity/manager.rs:184]: Client is also an Arc under the hood and designed (by tonic) to be cheap to clone

Comment (hzxuzhonghu) [reply src/identity/manager.rs:184]: Thanks for clarify

**Pull request merged**